fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductAnalysis.kt

```diff
@@ -300,13 +300,15 @@ private fun SettingsCard(
         Column {
             Spacer(modifier = Modifier.height(8.dp))
 
-            SwitchWithLabel(
-                checked = productRecommendationsEnabled,
-                onCheckedChange = onProductRecommendationsEnabledStateChange,
-                label = stringResource(R.string.review_quality_check_settings_recommended_products),
-            )
+            if (productRecommendationsEnabled != null) {
```

Comment:
optional nit (assuming the compiler doesn't still thing `productRecommendationsEnabled` is unsafe after the `let`):
```suggestion
            productRecommendationsEnabled?.let {
```

Reply:
I think if is a bit more expressive here, don't have a strong opinion either way